File: demo/messaging/Dockerfile.php
# cairn-chat: P2P messaging demo (PHP)
# Build from repo root: docker build -f demo/messaging/Dockerfile.php -t cairn-demo-messaging-php .

FROM php:8.3-cli-bookworm
WORKDIR /app

# Install composer
COPY --from=composer:2 /usr/bin/composer /usr/bin/composer

# Copy local cairn-p2p package
COPY packages/php/cairn-p2p /app/packages/php/cairn-p2p

# Copy demo files
COPY demo/messaging/php /app/demo/messaging/php

# Install dependencies
WORKDIR /app/demo/messaging/php
RUN composer install --no-interaction --no-dev 2>/dev/null || true

RUN useradd -r -s /bin/false cairn
USER cairn
ENTRYPOINT ["php", "cairn_chat.php"]

File: demo/messaging/Dockerfile.go
# cairn-chat: P2P messaging demo (Go)
# Build from repo root: docker build -f demo/messaging/Dockerfile.go -t cairn-demo-messaging-go .

# --- Builder stage ---
FROM golang:1.24-bookworm AS builder
WORKDIR /build
COPY . .
WORKDIR /build/demo/messaging/go
RUN go build -o cairn-chat .

# --- Runtime stage ---
FROM debian:bookworm-slim
RUN apt-get update && apt-get install -y --no-install-recommends ca-certificates && rm -rf /var/lib/apt/lists/*
RUN useradd -r -s /bin/false cairn
USER cairn
COPY --from=builder /build/demo/messaging/go/cairn-chat /usr/local/bin/cairn-chat
ENTRYPOINT ["cairn-chat"]
